packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -220,19 +220,28 @@ exports.sourceNodes = async (
         if (disallowedLinkTypes.includes(type)) return
         if (!url) return
         if (!type) return
+
+        // Apply any filters configured in gatsby-config.js. Filters
+        // can be any valid JSON API filter query string.
+        // See https://www.drupal.org/docs/8/modules/jsonapi/filtering
+        if (typeof filters === `object`) {
```

Comment:
Is there any way to refactor this into a function that can be reused inside and before `getNext`?

Reply:
The recursive getNext call looks like this

```
if (d.data.links && d.data.links.next) {
  data = await getNext(d.data.links.next, data)
}
```

so it gets the subsequent pages for the same entity type listing. The `next` links already contain the filters which were applied to the first page, so there's no need to add them again.

Reply:
:+1: Perfect! Yeah I came back around to this and that's fine.